src/main/java/ui/PanelMenuCreator.java

```diff
@@ -0,0 +1,133 @@
+                public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number arg2) {
+                    for (Node child : panelsScrollPane.getChildrenUnmodifiable()) {
+                        if (child instanceof ScrollBar) {
+                            ScrollBar scrollBar = (ScrollBar) child;
+                            if (scrollBar.getOrientation() == Orientation.HORIZONTAL
+                                    && scrollBar.visibleProperty().get()) {
+                                panelControl.scrollToCurrentlySelectedPanel();
+                                break;
+                            }
+                        }
+                    }
+                    panelControl.widthProperty().removeListener(this);
+                }
+            };
+            panelControl.widthProperty().addListener(listener);
+        });
+        createRight.setAccelerator(CREATE_RIGHT_PANEL);
+        return createRight;
+    }
+
+    public MenuItem closePanel(){
+        MenuItem closePanel = new MenuItem("Close");
+        closePanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Close");
+            panelControl.closeCurrentPanel();
+        });
+        closePanel.setAccelerator(CLOSE_PANEL);
+        return closePanel;
+    }
+
+    public MenuItem createAssigneePanel(){
+        MenuItem assigneePanel = new MenuItem("Self-assigned issues");
+        assigneePanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Self-assigned issues panel");
+            generatePanelWithNameAndFilter("Open issues and PR's",
+                    "is:open ((is:issue assignee:me) OR (is:pr author:me))");
+            panelControl.selectLastPanel();
+        });
+        return assigneePanel;
+    }
+
+    public MenuItem createRecentlyUpdatedPanel(){
+        MenuItem recentlyUpdatedPanel = new MenuItem("Recently Updated issues");
+        recentlyUpdatedPanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Recently Updated issues panel");
+            generatePanelWithNameAndFilter("Recently Updated issues", "assignee:me updated:<48");
+            panelControl.selectLastPanel();
+        });
+        return recentlyUpdatedPanel;
+    }
+
+    public MenuItem createMilestonePanel(){
+        MenuItem recentlyUpdatedPanel = new MenuItem("Current Milestone");
+        recentlyUpdatedPanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Current Milestone panel");
+            generatePanelWithNameAndFilter("Current Milestone", "milestone:curr sort:status");
+            panelControl.selectLastPanel();
+        });
+        return recentlyUpdatedPanel;
+    }
+
+    public FilterPanel generatePanelWithNameAndFilter(String panelName, String filterName){
+        FilterPanel panelAdded = panelControl.addPanelAt(panelControl.getPanelCount());
+        panelAdded.setPanelName(panelName);
+        panelAdded.setFilterByString(filterName);
+        return panelAdded;
+    }
+}
```

Comment:
Yup, `generatePanelWithNameAndFilter` should ideally be in `PanelControl.java` to avoid duplication. I'll put it there. The board auto creator class should then be able to use it!